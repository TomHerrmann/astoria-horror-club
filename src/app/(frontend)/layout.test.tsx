import { render, screen } from '@testing-library/react'
import RootLayout from './layout'
import { describe, expect, it, vi } from 'vitest'

vi.mock('next/font/google', () => ({
  Creepster: vi.fn().mockReturnValue({
    className: 'mock-creepster-font', // Mock className we can test for
  }),
}))

describe('RootLayout', () => {
  it('renders child content', () => {
    const { getByText } = render(
      <RootLayout>
        <div>Test Content</div>
      </RootLayout>,
    )
    screen.debug()

    expect(getByText('Test Content')).toBeDefined()
  })

  it('matches snapshot', () => {
    const { container } = render(
      <RootLayout>
        <div>Test Content</div>
      </RootLayout>,
    )
    expect(container).toMatchSnapshot()
  })
})
