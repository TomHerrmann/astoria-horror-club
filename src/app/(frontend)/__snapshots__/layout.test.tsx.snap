// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RootLayout > matches snapshot 1`] = `<div />`;
